### DISCUSSION TITLE
请问 Xray(M+F+H+K+A)+Caddy(N) 中有些域名为什么使用非 AES 算法？

### DISCUSSION THREAD

**Discussion by choicky** [lxhao61/integrated-examples#208] [Q&A] [answered]
例如，配置文件 https://github.com/lxhao61/integrated-examples/blob/main/Xray(M%2BF%2BH%2BK%2BA)%2BCaddy(N)/2_caddy.json 写着 ：

```
本示例使用 REALITY Vision 配置（h2r.example.com）、非 AES 算法的密码套件配置（t2n.example.com）、非 AES 算法的密码套件配置（h2y.example.com）、CDN 流量中转（cdn.example.com）来避免被封。
```
我有两个疑问：

1. `t2n.example.com` 和  `h2y.example.com` 为什么需要非AES算法的密码套件？尤其是 `h2y.example.com`  为什么需要非AES算法的密码套件？

2. 我看配置文件的具体内容，感觉只限定了   `h2y.example.com`  使用非AES的密码套件，没有限定 `t2n.example.com` 。是漏了，还是笔误了（例如应该是 `t2n` 而非 `h2y` ）？

**Comment by lxhao61** [comment, marked as answer]
回答：
1、使用 TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 密码套件（非 AES 算法）可以防止被封，用不同域名对应只是来区分应用的。
2、域名 t2n.example.com 对应的是 Trojan+TCP+TLS 应用，此限定在 Xray 配置中设置，不需在 Caddy 配置中设置。

**Reply by choicky** [reply]
多谢解答。不过，本配置中，没看到 `h2y.example.com` 对应的后端服务，所以我不太理解哪种类型的后端服务搭配非AES的密码套件比较好。

**Reply by lxhao61** [reply]
晕，看配置下边备注！除了Xray 的 REALITY Vision 配置、XTLS Vision 配置、REALITY 配置外，其它应用搭配非 AES 的密码套件比较好。
